geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/SimpleFeatureConverterFactory.scala

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2014 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.convert
+
+
+import javax.imageio.spi.ServiceRegistry
+
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.slf4j.Logging
+import org.locationtech.geomesa.convert.Transformers.{EvaluationContext, Expr}
+import org.locationtech.geomesa.feature.AvroSimpleFeatureFactory
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.JavaConversions._
+import scala.util.{Failure, Success, Try}
+
+trait Field {
+  def name: String
+  def transform: Transformers.Expr
+  def eval(args: Any*)(implicit ec: EvaluationContext): Any = transform.eval(args: _*)
+}
+
+case class SimpleField(name: String, transform: Transformers.Expr) extends Field
+
+trait SimpleFeatureConverterFactory[I] {
+
+  def canProcess(conf: Config): Boolean
+
+  def canProcessType(conf: Config, name: String) = Try { conf.getString("type").equals(name) }.getOrElse(false)
+
+  def buildConverter(sft: SimpleFeatureType, conf: Config): SimpleFeatureConverter[I]
+
+  def buildFields(fields: Seq[Config]): IndexedSeq[Field] =
+    fields.map { f =>
+      val name = f.getString("name")
+      val transform = Transformers.parseTransform(f.getString("transform"))
+      SimpleField(name, transform)
+    }.toIndexedSeq
+
+  def buildIdBuilder(t: String) = Transformers.parseTransform(t)
+
+}
+
+object SimpleFeatureConverters {
+  val providers = ServiceRegistry.lookupProviders(classOf[SimpleFeatureConverterFactory[_]]).toList
+
+  def build[I](sft: SimpleFeatureType, conf: Config) = {
+    val converterConfig = conf.getConfig("converter")
+    providers
+      .find(_.canProcess(converterConfig))
+      .map(_.buildConverter(sft, converterConfig).asInstanceOf[SimpleFeatureConverter[I]])
+      .getOrElse(throw new IllegalArgumentException("Cannot find factory"))
```

Comment:
Could we add some detail in the exception?  Adding the SFT typename and the classpath searched would be awesome.